File: go/coin/dnmcprgrmng/cnsm/mztn/solution_test.go
package cnsm

import "testing"

type testCase struct {
	target   int
	numbers  []int
	expected bool
}

func Test_correctness(t *testing.T) {
	tests := []testCase{
		{0, []int{1, 2, 3}, true},
		{1, []int{1}, true},
		{1, []int{2}, false},
		{3, []int{1, 2}, true},
		{7, []int{2, 3}, true},
		{7, []int{5, 3, 4, 7}, true},
		{7, []int{2, 4}, false},
		{8, []int{2, 3, 5}, true},
	}
	for _, test := range tests {
		if actual := canSum(test.target, test.numbers); actual != test.expected {
			t.Error("expected that the", test.target, "sum can be built from", test.numbers)
			t.Fail()
		}
	}
}

func Test_performance(t *testing.T) {
	canSum(300, []int{7, 14})
}

File: csharp/coin/dnmcprgrmng/cnsm/mztn/SolutionTest.cs
using NUnit.Framework;

namespace dnmcprgrmng.mztn.brtfrc
{
    public class SolutionTest
    {
        private Solution solution;

        [SetUp]
        public void BeforeEachTest()
        {
            solution = new Solution();
        }

        [TestCase(0, new int[] { 1, 2, 3 }, true)]
        [TestCase(1, new int[] { 1 }, true)]
        [TestCase(1, new int[] { 2 }, false)]
        [TestCase(2, new int[] { 3, 2 }, true)]
        [TestCase(7, new int[] { 2, 3 }, true)]
        [TestCase(7, new int[] { 5, 3, 4, 7 }, true)]
        [TestCase(7, new int[] { 2, 4 }, false)]
        [TestCase(8, new int[] { 2, 3, 5 }, true)]
        public void Correctness(int target, int[] numbers, bool expected)
        {
            Assert.That(solution.CanSum(target, numbers), Is.EqualTo(expected));
        }

        [TestCase(3000, new int[] { 7, 14 }, false)]
        public void Performance(int target, int[] numbers, bool expected)
        {
            Assert.That(solution.CanSum(target, numbers), Is.EqualTo(expected));
        }
    }
}
